core/src/Kokkos_Core.hpp

```diff
@@ -234,6 +301,54 @@ class KOKKOS_ATTRIBUTE_NODISCARD ScopeGuard {
   ScopeGuard(const ScopeGuard&)            = delete;
 };
 
+#else  // ifndef KOKKOS_ENABLE_DEPRECATED_CODE3
```

Comment:
In case we ever change the initialization logic, we would have to remember to update the code in two places. I'd suggest delegating the warnings/errors to a function or macro then depending on the state of `KOKKOS_ENABLE_DEPRECATED_CODE3` aborting or warning.

Reply:
But the first branch would go away in 4.0, I just split it completely out so I don't have ifdefs in every second line.

Reply:
ok makes sense, not too worried about it then